packages/grid/_modules_/grid/components/default-footer.tsx

```diff
@@ -24,12 +25,26 @@ export const DefaultFooter = React.forwardRef<HTMLDivElement, DefaultFooterProps
       return null;
     }
 
+    const isPaginationAvailable = !!paginationComponent;
+    const showRowCount = !options.hideFooterRowCount && !isPaginationAvailable && (
+      <RowCount rowCount={totalRowCount} />
+    );
+    const showSelectedRowCount = !options.hideFooterSelectedRowCount && (
+      <SelectedRowCount selectedRowCount={selectedRowCount} />
+    );
+    const justifyItemsRight =
+      !selectedRowCount && !options.hideFooterSelectedRowCount && isPaginationAvailable;
+
     return (
-      <GridFooter ref={ref}>
-        {!options.hideFooterRowCount && <RowCount rowCount={totalRowCount} />}
-        {!options.hideFooterSelectedRowCount && (
-          <SelectedRowCount selectedRowCount={selectedRowCount} />
-        )}
+      <GridFooter
+        ref={ref}
+        className={classnames({
+          'MuiDataGrid-footer-pagination-visible': isPaginationAvailable,
+          'MuiDataGrid-footer-justifyContent-right': justifyItemsRight,
```

Comment:
Adding a hypen for separating internal components was one thing, doing it for naming is another. 

```suggestion
          'MuiDataGrid-footer-justifyContentRight': justifyItemsRight,
```

Reply:
Instead of "right" using "end" would be better for RTL.

Reply:
What if we render an empty div instead so we don't have to apply a custom class name?

Reply:
Or we could also inverse the direction of the flex container. That could be even better if that's what most people would want, not sure.

Reply:
I was thinking of rendering an empty div as well but I didn't like how it looked. Maybe we can reintroduce it. The benefit is that if we add a single empty div around the `row selected` we don't need to add any additional CSS login / styles